pkg/syncer/shared/finalizer.go

```diff
@@ -73,31 +72,26 @@ func EnsureUpstreamFinalizerRemoved(ctx context.Context, gvr schema.GroupVersion
 	}
 	upstreamObj.SetFinalizers(desiredFinalizers)
 
-	//  TODO(jmprusi): This code block will be handled by the syncer virtual workspace, so we can remove it once
-	//                 the virtual workspace syncer is integrated
+	// TODO(davidfestal): The following code block should be removed as soon as the Advanced Scheduling feature is
```

Comment:
nit: maybe you all have no problem remembering all of your TODOs but I would suggest making this code panic or not compile or something to make it easier to find what you need to change when scheduling is removed :)